lms/djangoapps/ccx/migrations/0004_seed_forum_roles_in_ccx_courses.py

```diff
@@ -20,38 +23,40 @@ def seed_forum_roles_for_existing_ccx(apps, schema_editor):
     """
     CustomCourseForEdX = apps.get_model("ccx", "CustomCourseForEdX")
     Role = apps.get_model("django_comment_common", "Role")
-    list_ccx = CustomCourseForEdX.objects.all()
-    for ccx in list_ccx:
+    Permission = apps.get_model("django_comment_common", "Permission")
+    db_alias = schema_editor.connection.alias
+
+    for ccx in CustomCourseForEdX.objects.using(db_alias).all():
         if not ccx.course_id or ccx.course_id.deprecated:
             # prevent migration for deprecated course ids or invalid ids.
             continue
-        ccx_locator = CCXLocator.from_course_locator(ccx.course_id, unicode(ccx.id))
-        # Seed forum roles.
-        seed_permissions_roles(ccx_locator)
-        # Make CCX coach forum admin.
-        admin_role = Role.objects.get(course_id=ccx_locator, name=FORUM_ROLE_ADMINISTRATOR)
-        ccx.coach.roles.add(admin_role)
 
+        ccx_locator = CCXLocator.from_course_locator(ccx.course_id, unicode(ccx.id))
 
-def remove_forum_roles_for_existing_ccx(apps, schema_editor):
-    """
-    Remove seeded forum roles from respective CCX(s).
+        # Create forum roles.
+        role_query_set = Role.objects.using(db_alias)
+        admin_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_ADMINISTRATOR, course_id=ccx_locator)
+        moderator_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_MODERATOR, course_id=ccx_locator)
+        community_ta_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_COMMUNITY_TA, course_id=ccx_locator)
+        student_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_STUDENT, course_id=ccx_locator)
 
-    Arguments:
-        apps (Applications): Apps in edX platform.
-        schema_editor (SchemaEditor): For editing database schema i.e create, delete field (column)
+        # Add permissions for each role.
+        permission_query_set = Permission.objects.using(db_alias)
+        for name in _ADMINISTRATOR_ROLE_PERMISSIONS:
```

Comment:
More typos here; the leading underscores are gone.  I pushed a fix for these as well.